### DISCUSSION TITLE
Menggunakan PHPNuxBill dengan bantuan Docker

### DISCUSSION THREAD

**Discussion by animegasan** [hotspotbilling/phpnuxbill#26] [General]
Sebelumnya kepada mas  Ibnu Maksum selaku pembuat aplikasi,
Saya izin membagikan images PHPNuxBill agar dapat di gunakan dengan bantuan Docker.

Dokumentasi dan cara instalasi ada di halaman [ini](https://github.com/animegasan/phpnuxbill)

**Comment by ibnux** [comment]
Mantap 

Nanti saya link ke reponya

**Reply by animegasan** [reply]
Okey, selanjutnya mungkin saya akan monitoring bila ada update dari mas ibnu

**Reply by umblox** [reply]
permisi om, maaf sebelum nya, saya pemula. yang ingin saya tanyakan, apakah bisa berjalan dengan hp android saja? saya sudah mencoba menggunakan awebserver dan juga ksweb, juga mencoba saya up ke hosting. ketiga cara bisa sampai tahap install, login dashboard phpnuxbill, menghubungkan mikrotik semua bisa, namun saat mencoba generate voucher, di mikrotik tidak muncul voucher tersebut. dan juga saat import hotspot, yang masuk malah cuma user profil nya saja sedangkan user nya tidak. kalo untuk pppoe bisa terload beserta user nya.

**Reply by ibnux** [reply]
@umblox tidak dianjurkan di Android, karena tidak ada Cronjobnya

beberapa ada yang beli STB Telxom yang diubah ke Linux, dan diinstall kesitu

coba dipelajari cara kerjanya

https://github.com/hotspotbilling/phpnuxbill/wiki/How-It-Works---Cara-kerja

**Reply by umblox** [reply]
di apk ksweb ada cron.job dan bisa berjalan om. barusan saya juga mencoba phpmixbill juga sama kendala gak bisa sinkron voucher user hotspot nya dengan mikrotik, yg saya penasaran kenapa user profil hotspot nya bisa, juga pppoe pun bisa sinkron.
ok saya ada stb bekas buat openwrt nanti beli memory lagi buat linux ato openwrt om. 🙏

**Reply by ibnux** [reply]
@umblox sip kalo ada cronjob

cara kerja phpnuxbill harus tahu dulu

Voucher memang ngga masuk ke mikrotik

Cuma bisa di Linux. Kecuali openwrt bisa PHP mysql

**Reply by renoadi06** [reply]
untuk pasang di hosting, ada kendala ga om waktu menambahnkan NAS / mikrotiknya ?

**Comment by umblox** [comment]
@ibnux baik om udah sedikit paham cara kerja nya.

 udah input walled garden localhost nya, juga walled garden untuk hosting web server nya, tapi ga bisa di buka di hp pelanggan om. localhost ga bisa buka, kalo di hosting keterangan acces denied by mikrotik kalo di buka pelanggan yg belom login

**Reply by ibnux** [reply]
Nah kalo itu sudah teknikal

Bantuan teknis berbayar 😁

**Comment by umblox** [comment]
@ibnux sekarang sudah bisa om, saya tambah wg -ip list hosting, user udah bisa daftar, dapet otp dan login bisa login, udah masukin voucher, dpt notif juga. tapi gada akses internet 😅😆

**Reply by ibnux** [reply]
Usernya harus login juga di Mikrotik hotspot login

**Reply by umblox** [reply]
@ibnux berarti vouchernya di masukin di dua tempat? login page mikrotik dan di dashboard beranda phpnuxbill?
edit : oh sudah paham. user login pake akun phpnuxbill juga di loginpage mikrotik. hehe 
sekarang tinggal cari loginpage yg include phpnuxbill dong, wkwk

**Comment by ibnux** [comment]
@animegasan perlu ditambah ekstensi php-zip di dockernya

Kalo bisa ambil script phpnuxbillnya dengan git clone, jadi ngga perlu update terus

Ada issue nya
https://github.com/hotspotbilling/phpnuxbill/issues/58

**Reply by animegasan** [reply]
Untuk file bisa saya pakai git-clone, tetapi untuk extension harus terinstall manual ke host. Jadi besok saya akan perbaharui nya. Saya lupa menambahkan extension zipnya.

**Comment by animegasan** [comment]
Kalau boleh tahu, extension apa saja yang dipakai untuk PHPNuxBill?
Untuk image terdahulu hanya terpasang ini

> php81
> php81-fpm
> php81-gd
> php81-mbstring
> php81-mysqli
> php81-session
> php81-pdo
> php81-pdo_mysql

**Reply by ibnux** [reply]
Bisa di set di php74 aja?
Kayanya lebih stabil di 7.4 untuk saat ini


Untuk pdf butuh
mbstring
zlib
bcmath
xml

zip
curl

**Reply by ibnux** [reply]
Biasanya saya install di Linux gini

apt-get install php74-{fpm,cli,pdo,gd,mbstring,mysqlnd,opcache,xml,zip,curl,mysql,xml,zip,imap,json,sqlite,imap,calendar,ctype,dom,exif,fileinfo,ftp,gettext,iconv,phar,posix,readline,shmop,simplexml,sockets,sysvmsg,sysvsem,sysvshm,tokenizer,wddx,xmlreader,xmlwriter,xsl,imagick,intl}

**Reply by animegasan** [reply]
> Bisa di set di php74 aja? Kayanya lebih stabil di 7.4 untuk saat ini
> 
> Untuk pdf butuh mbstring zlib bcmath xml
> 
> zip curl

Bisa, berarti saya turunkan juga os nya biar menyesuaikan ke php7 soalnya yang kemarin di buat latest

> Biasanya saya install di Linux gini
> 
> apt-get install php74-{fpm,cli,pdo,gd,mbstring,mysqlnd,opcache,xml,zip,curl,mysql,xml,zip,imap,json,sqlite,imap,calendar,ctype,dom,exif,fileinfo,ftp,gettext,iconv,phar,posix,readline,shmop,simplexml,sockets,sysvmsg,sysvsem,sysvshm,tokenizer,wddx,xmlreader,xmlwriter,xsl,imagick,intl}

OK, semua extensi yang diperlukan saya install semua

**Reply by tuyewxx** [reply]
Docker image nya belum di update bang?

**Comment by renoadi06** [comment]
ijin nanya om @umblox , pasang di hosting ada kendala ga pas menambahkan NAS / mikrotiknya, boleh tanya pake hosting apa? waktu itu saya coba pake hosting ****** yang murah buat coba2

> permisi om, maaf sebelum nya, saya pemula. yang ingin saya tanyakan, apakah bisa berjalan dengan hp android saja? saya sudah mencoba menggunakan awebserver dan juga ksweb, juga mencoba saya up ke hosting. ketiga cara bisa sampai tahap install, login dashboard phpnuxbill, menghubungkan mikrotik semua bisa, namun saat mencoba generate voucher, di mikrotik tidak muncul voucher tersebut. dan juga saat import hotspot, yang masuk malah cuma user profil nya saja sedangkan user nya tidak. kalo untuk pppoe bisa terload beserta user nya.
